common/web/nunjucks/index.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2021 Choko (choko@curioswitch.org)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import CurioNunjucksEnvironment from './src/CurioNunjucksEnvironment';
+
+const nunjucks = require('nunjucks');
```

Comment:
```suggestion
import nunjucks;
```

Since we have the types now does this work?

Reply:
Tried it. Didn't work, since `Environment` is read-only and can't be overwritten.

```
TSError: ⨯ Unable to compile TypeScript:
common/web/nunjucks/index.ts(28,10): error TS2540: Cannot assign to 'Environment' because it is a read-only property.
```

Reply:
A type error is often a code smell, and I think in this case it's correct - modifying a global variable, especially in JS with the many different patterns of package loaders that often get in the way of such patterns.

I notice in the README we mention explicitly the simple API isn't supported and when I tried running a unit test with it it indeed didn't work, presumably because the internal `Environment` class gets captured by a callback before we can modify it. 

So how about we just export only the modified environment instead of reexporting `nunjuck`? Then the caveat about the simple API goes away since we don't export it. And we wouldn't need to modify the global.

Reply:
I actually tried only exporting `Environment` before but it presented a problem, during implementation. Here's an example of what I mean:
```
import CurioNunjucksEnvironment from '@curiostack/nunjucks';
const nunjucks = require('nunjucks');

const env = new CurioNunjucksEnvironment(nunjucks.FileSystemLoader('some/path'));
// Need both the import and require logic for this ^
```

Of course I could modify `CurioNunjucksEnvironment` with constructors, which don't need the `nunjucks` classes, or export those as well. But that means we'll be doing more work to maintain it, with not much benefit.

Due to this reason I would prefer to leave it as is now.

Reply:
So I guess that means the default when just calling `new CurioNunjucksEnvironment()` which I think has default logic for setting up template loading, doesn't work well in most cases? In that case, I'd go ahead and reexport the methods that make sense, the maintenance effort isn't too big. It makes clear what the API of this module is - we prefer less gotchas such as the one we have in the README right now about support for the simple API, and the questions around mutating the global variable arbitrarily.

Reply:
Fair point. I'll try to add all the classes to the export which actually matter, for working with the `Environment`.

Btw which would fit better?
```
import { Environment, FileSystemLoader } from '@curiostack/nunjucks';
// OR
import nunjucks from '@curiostack/nunjucks';
// ^ has only the necessary classes and methods, excluding the Simple API
```
I'm leaning more toward the first, since that instinctively feels a bit different from the base `nunjucks` implementation, giving a hint to the implementer, to check which classes and methods are available.

Reply:
I think either `{ Environment, FileSystemLoader }`, or `Environment, { FileSystemLoader }`. It depends on how useful just `new Environment()` is, if that has use cases, I'd let the default export capture the simplest case and have the named exports be the less simple stuff.

Reply:
I think `new Environment()` is useful enough. Since it defaults to the project/module directory as the context. Which is a reasonable default.
I'll go with `Environment, { FileSystemLoader }`.